dev/com.ibm.ws.security.acme_fat/fat/src/com/ibm/ws/security/acme/fat/AcmeSwapDirectoriesTest.java

```diff
@@ -0,0 +1,488 @@
+					if (tok.hasMoreTokens()) tok.nextToken();
+					if (tok.hasMoreTokens()) secondDirURI = tok.nextToken();
+				}
+			} catch (Exception e) {
+				fail(e.getMessage()); 
+			}
+			assertNotSame(firstDirURI, secondDirURI);
+			assertEquals(caContainer2.getAcmeDirectoryURI(false), secondDirURI);
+			Log.info(this.getClass(), testName.getMethodName(), "TEST 3: FINISH");
+
+			/***********************************************************************
+			 * 
+			 * TEST 4: Stop the server, change the directoryURI, and start the server.
+			 * We should renew the certificate.
+			 * 
+			 **********************************************************************/			
+			Log.info(this.getClass(), testName.getMethodName(), "TEST 4: START");
+			server.stopServer();
+			AcmeFatUtils.configureAcmeCA(server, caContainer, ORIGINAL_CONFIG, false, DOMAINS_1);
+			server.startServer();
+			AcmeFatUtils.waitForAcmeAppToStart(server);
+			AcmeFatUtils.waitForAcmeToCreateCertificate(server);
+			AcmeFatUtils.waitForSslEndpoint(server);
+
+			/*
+			 * Verify that the server is now using a certificate signed by the
+			 * new CA.
+			 */
+			Certificate[] certificates3 = AcmeFatUtils.assertAndGetServerCertificate(server, caContainer);
+
+			BigInteger serial3 = ((X509Certificate) certificates3[0]).getSerialNumber();
+			assertFalse("Expected new certificate after adding new domain.", serial2.equals(serial3));
+			Log.info(this.getClass(), testName.getMethodName(), "TEST 4: FINISH");
+			
+			/***********************************************************************
+			 * 
+			 * TEST 5: Check that the ACME historical file was updated. 
+			 * 
+			 **********************************************************************/			
+			Log.info(this.getClass(), testName.getMethodName(), "TEST 5: START");
+			String thirdDirURI = null;
+			try {
+			    BufferedReader br = new BufferedReader(new FileReader(file));
+			    String line, lastLine = null; 
+				while ((line = br.readLine()) != null && !line.isEmpty()) {
+					lastLine=line;
+				}
+				br.close();
+				if (lastLine != null && !lastLine.isEmpty()) {
+					StringTokenizer tok = new StringTokenizer(lastLine);
+					if (tok.hasMoreTokens()) tok.nextToken();
+					if (tok.hasMoreTokens()) tok.nextToken();
+					if (tok.hasMoreTokens()) thirdDirURI = tok.nextToken();
+				}
+			} catch (Exception e) {
+				fail(e.getMessage()); 
+			}
+			assertNotSame(secondDirURI, thirdDirURI);
+			assertEquals(caContainer.getAcmeDirectoryURI(false), thirdDirURI);
+			Log.info(this.getClass(), testName.getMethodName(), "TEST 5: FINISH");
+		} finally {
+			/*
+			 * Stop the server.
+			 */
+			server.stopServer("CWPKI2038W");
+		}
+	}
+	/**
+	 * This test will verify that the certificate is refreshed
+	 * when the ACME file is unreadable or unwritable.
+	 * 
+	 * @throws Exception
+	 *             If the test failed for some reason.
+	 */
+	@Test
+    @AllowedFFDC(value = { "java.io.FileNotFoundException", "java.io.IOException" })
```

Comment:
I can't change the permissions while the server is down because when a server starts, a new server directory is created, eg. com.ibm.ws.security.acme.fat.simple-21-05-2020-16-03-05. Thus, any acme file i would have created would be down a different path.
Because of this, I can't use different directoryURIs to trigger a config update because that would refresh the certificate, and I'm trying to test that an unreadable/unwritable acme file triggers a refresh. So I use domain changes that should not trigger a refresh on their own. 
1. Start the server and wait for initial certificate
2. Set the acme file to be unreadable. The next time we try to read it, we should force refresh the certificate.
3. Update the config with a change that does NOT force refresh and wait for the certificate to refresh.
4. Delete the acme file and set the acme directory to be unreadable.
5. Change the directoryURI. We won’t be able to write the acme file, but we should refresh the certificate.

Reply:
I thought that the acme.fat.simple directory remained, it just archived the contents at the end of the previous test (this is why we clean up the keystores at the end of each test). Perhaps I am wrong. As long as you are certain this is testing what you believe it is.